### PR TITLE
[#2879] feat(docs): Add doc for Doris catalog

### PR DESCRIPTION
**Pull request by zhoukangcn**
### What changes were proposed in this pull request?

Add doc for Doris catalog

### Why are the changes needed?

Fix: #2879

### Does this PR introduce _any_ user-facing change?

N/A

### How was this patch tested?
N/A

### CODE REVIEW

File: docs/jdbc-doris-catalog.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+:::info
+MySQL doesn't support Gravitino `Fixed` `Struct` `List` `Map` `Timestamp_tz` `IntervalDay` `IntervalYear` `Union` `UUID` type.
+Meanwhile, the data types other than listed above are mapped to Gravitino **[Unparsed Type](./manage-relational-metadata-using-gravitino.md#unparsed-type)** that represents an unresolvable data type since 0.5.0.
+:::
+
+#### Table column auto-increment
+
+:::note
+Doris does not support auto-increment column before version 2.1.0
+So, Gravitino did not support auto-increment column before version 0.5.0
```
Comment (yuqi1129) [inline docs/jdbc-doris-catalog.md:108]: The causal relationship between `Doris does not support auto-increment column before version 2.1.0` and `Gravitino did not support auto-increment column before version 0.5.0` seems to be very odd. 

I think we'd better change it to `Gravitino did not support create an auto-increment column for Doris table if the version of Doris version is below 2.1.0`

File: docs/jdbc-doris-catalog.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+- `UpdateColumnType`
+- `UpdateColumnPosition`
+- `UpdateColumnComment`
+- `SetProperty`
+
+:::info
+ - Not all table alteration operations can be processed in batch. 
+ - Schema change, such as add/modify/drop columns can be processed in batch
+ - Can modify multi column comments at same time
+ - You can't modify the column type and column comment at the same time
```
Comment (yuqi1129) [inline docs/jdbc-doris-catalog.md:172]: Can you describe the asynchronous process of schema change here?

File: docs/jdbc-doris-catalog.md (hunk 1 above)
Comment (zhoukangcn) [reply docs/jdbc-doris-catalog.md]: Docs is updated 
```
Gravitino does not support the creation of an auto-increment column for a Doris table now.
```

I have not yet implemented the feature to determine whether auto-increment is supported based on the version of Doris.

File: docs/jdbc-doris-catalog.md (hunk 2 above)
Comment (zhoukangcn) [reply docs/jdbc-doris-catalog.md:171]: done

File: docs/jdbc-doris-catalog.md (hunk 1 above)
Comment (yuqi1129) [reply docs/jdbc-doris-catalog.md]: OK

File: docs/assets/webui/props-doris.png
Comment (yuqi1129) [inline docs/assets/webui/props-doris.png:1]: Can you provide a clearer screenshot here, the picture you provided is just a typical JDBC catalog creation page.
it seems unrelated to Doris.

File: docs/jdbc-doris-catalog.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+:::caution
+Gravitino saves some system information in schema and table comment, like `(From Gravitino, DO NOT EDIT: gravitino.v1.uid1078334182909406185)`, please don't change or remove this message.
+:::
+
+## Catalog
+
+### Catalog capabilities
+
+- Gravitino catalog corresponds to the Doris instance.
+- Supports metadata management of Doris (1.2.x).
```
Comment (yuqi1129) [inline docs/jdbc-doris-catalog.md:28]: Does version `1.2.x` is the mostly used version currently?

File: docs/jdbc-doris-catalog.md [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+    Indexes.of(IndexType.PRIMARY_KEY, "PRIMARY", new String[][]{{"id"}})
+}
+```
+
+</TabItem>
+</Tabs>
+
+### Table operations
+
+Refer to [Manage Relational Metadata Using Gravitino](./manage-relational-metadata-using-gravitino.md#table-operations) for more details.
```
Comment (yuqi1129) [inline docs/jdbc-doris-catalog.md:152]: Please xxxx

File: docs/assets/webui/props-doris.png
Comment (zhoukangcn) [reply docs/assets/webui/props-doris.png:1]: done. Please take a look again.

File: docs/jdbc-doris-catalog.md (hunk 3 above)
Comment (zhoukangcn) [reply docs/jdbc-doris-catalog.md:28]: Doris 2.x version utilizes a vectorized engine. Xiaomi is still in the testing phase and has not officially used it yet. 

Also, there are updates in Doris 2 metadata, such as auto-increment column. 

We could consider supporting Doris 2 later

Comment (yuqi1129) [reply docs/jdbc-doris-catalog.md:28]: OK, let's change it after we have thoroughly tested other versions.

**Comment by yuqi1129** [comment]
@jerryshao 
Do you need to take a look?

**Comment by jerryshao** [comment]
> @jerryshao Do you need to take a look?

Yeah, I will.

File: docs/jdbc-doris-catalog.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+
+### Schema capabilities
+
+- Gravitino's schema concept corresponds to the Doris database.
+- Supports creating schema.
+- Supports dropping schema.
+
+### Schema properties
+
+- Support schema properties
```
Comment (jerryshao) [inline docs/jdbc-doris-catalog.md:71]: What kind of schema properties do we support currently?

File: docs/jdbc-doris-catalog.md [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+| `Double`       | `Double`   |
+| `Decimal`      | `Decimal`  |
+| `Date`         | `Date`     |
+| `Timestamp`    | `Datetime` |
+| `VarChar`      | `VarChar`  |
+| `FixedChar`    | `Char`     |
+| `String`       | `String`   |
+
+:::info
+MySQL doesn't support Gravitino `Fixed` `Struct` `List` `Map` `Timestamp_tz` `IntervalDay` `IntervalYear` `Union` `UUID` type.
```
Comment (jerryshao) [inline docs/jdbc-doris-catalog.md:104]: MySQL or Doris?

File: docs/jdbc-doris-catalog.md [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+license: "Copyright 2024 Datastrato Pvt Ltd.
+This software is licensed under the Apache License version 2."
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+## Introduction
+
+Gravitino provides the ability to manage Doris metadata.
```
Comment (jerryshao) [inline docs/jdbc-doris-catalog.md:17]: I would suggest change Doris to Apache Doris here to avoid ASF issue.

File: docs/jdbc-doris-catalog.md [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+#### Table column auto-increment
+
+:::note
+Gravitino does not support the creation of an auto-increment column for a Doris table now.
+:::
+
+
+### Table properties
+
+Doris supports table properties, and you can set them in the table properties.
```
Comment (jerryshao) [inline docs/jdbc-doris-catalog.md:117]: What kind of table properties do we support?

File: docs/jdbc-doris-catalog.md [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+- `DeleteColumn`
+- `UpdateColumnType`
+- `UpdateColumnPosition`
+- `UpdateColumnComment`
+- `SetProperty`
+
+:::info
+ - Not all table alteration operations can be processed in batch. 
+ - Schema change, such as add/modify/drop columns can be processed in batch
+ - Can modify multi column comments at same time
```
Comment (jerryshao) [inline docs/jdbc-doris-catalog.md:174]: Support modify multiple column comments at the same time.

File: docs/jdbc-doris-catalog.md [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+ - You can't modify the column type and column comment at the same time
```
Comment (jerryshao) [inline docs/jdbc-doris-catalog.md:175]: Doesn't support modify...

File: docs/jdbc-doris-catalog.md (hunk 7 above)
Comment (zhoukangcn) [reply docs/jdbc-doris-catalog.md:17]: Do you mean all `Doris` in the document, or just change at the beginning of the document?

Comment (jerryshao) [reply docs/jdbc-doris-catalog.md:17]: I think change at the beginning should be enough.

File: docs/jdbc-doris-catalog.md (hunk 5 above)
Comment (zhoukangcn) [reply docs/jdbc-doris-catalog.md:71]: docs updated. `Support schema properties, includes Doris database properties and user-defined properties.`

**Comment by zhoukangcn** [comment]
@jerryshao Doc has been updated. Please help to review again, thanks.

File: docs/jdbc-doris-catalog.md [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+- Supports table index.
+- Supports [column default value](./manage-relational-metadata-using-gravitino.md#table-column-default-value).
+
+:::note
+Doris 2.x supports more advanced metadata features such as auto-increment column. We will be adding support for it in the future.
+:::
+
+### Catalog properties
+
+You can pass to a Doris data source any property that isn't defined by Gravitino by adding `gravitino.bypass` prefix as a catalog property. For example, catalog property `gravitino.bypass.maxWaitMillis` will pass `maxWaitMillis` to the data source property.
```
Comment (jerryshao) [inline docs/jdbc-doris-catalog.md:38]: Actually it should be "gravitino.bypass.", right?

Comment (jerryshao) [reply docs/jdbc-doris-catalog.md:38]: Let me help you polish the doc.

File: docs/jdbc-doris-catalog.md [hunk 12]
```diff
@@ -0,0 +1,176 @@
...
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+## Introduction
+
+Gravitino provides the ability to manage [Apache Doris](https://doris.apache.org/) metadata through JDBC connection..
+
+:::caution
+Gravitino saves some system information in schema and table comment, like
```
Comment (justinmclean) [inline docs/jdbc-doris-catalog.md:20]: comment -> comments

File: docs/jdbc-doris-catalog.md [hunk 13]
```diff
@@ -0,0 +1,176 @@
...
+
+### Schema capabilities
+
+- Gravitino's schema concept corresponds to the Doris database.
+- Supports creating schema.
+- Supports dropping schema.
+
+### Schema properties
+
+- Support schema properties, includes Doris database properties and user-defined properties.
```
Comment (justinmclean) [inline docs/jdbc-doris-catalog.md:71]: includes -> including

File: docs/jdbc-doris-catalog.md [hunk 14]
```diff
@@ -0,0 +1,176 @@
...
+| `Double`       | `Double`   |
+| `Decimal`      | `Decimal`  |
+| `Date`         | `Date`     |
+| `Timestamp`    | `Datetime` |
+| `VarChar`      | `VarChar`  |
+| `FixedChar`    | `Char`     |
+| `String`       | `String`   |
+
+Doris doesn't support Gravitino `Fixed` `Struct` `List` `Map` `Timestamp_tz` `IntervalDay` `IntervalYear` `Union` `UUID` type.
+The data types other than listed above are mapped to Gravitino **[Unparsed Type](./manage-relational-metadata-using-gravitino.md#unparsed-type)**
```
Comment (justinmclean) [inline docs/jdbc-doris-catalog.md:105]: than listed above -> than those listed above

File: docs/jdbc-doris-catalog.md [hunk 15]
```diff
@@ -0,0 +1,176 @@
...
+that represents an unresolvable data type since 0.5.0.
+
+#### Table column auto-increment
+
+Unsupported for now.
+
+### Table properties
+
+- Doris supports table properties, and you can set them in the table properties.
+- Only supports Doris table properties, doesn't support user-defined properties.
```
Comment (justinmclean) [inline docs/jdbc-doris-catalog.md:115]: ", " -> " and "

File: docs/jdbc-doris-catalog.md [hunk 16]
```diff
@@ -0,0 +1,176 @@
...
+- `DeleteColumn`
+- `UpdateColumnType`
+- `UpdateColumnPosition`
+- `UpdateColumnComment`
+- `SetProperty`
+
+Please be aware that:
+
+ - Not all table alteration operations can be processed in batch.
+ - Schema change, such as add/modify/drop columns can be processed in batch.
```
Comment (justinmclean) [inline docs/jdbc-doris-catalog.md:170]: Schema change -> Schema changes
batch -> batches

File: docs/jdbc-doris-catalog.md [hunk 17]
```diff
@@ -0,0 +1,176 @@
...
+ - Supports modify multiple column comments at the same time.
+ - Doesn't support modify the column type and column comment at the same time.
```
Comment (justinmclean) [inline docs/jdbc-doris-catalog.md:172]: modify -> modifying

File: docs/jdbc-doris-catalog.md [hunk 18]
```diff
@@ -0,0 +1,176 @@
...
+Please refer to
+[Manage Relational Metadata Using Gravitino](./manage-relational-metadata-using-gravitino.md#schema-operations) for more details.
+
+## Table
+
+### Table capabilities
+
+- Gravitino's table concept corresponds to the Doris table.
+- Supports index.
+- Supports [column default value](./manage-relational-metadata-using-gravitino.md#table-column-default-value) and [auto-increment](./manage-relational-metadata-using-gravitino.md#table-column-auto-increment)..
```
Comment (jerryshao) [inline docs/jdbc-doris-catalog.md:84]: You mentioned that we don't support auto-increment for now in the below, but here it says it supports auto-increment, so which one is correct? @zhoukangcn @yuqi1129

Comment (zhoukangcn) [reply docs/jdbc-doris-catalog.md:84]: Sorry, It's a mistake here. I will update doc today

**Review by jerryshao — APPROVED** [review]
